### PR TITLE
Remove bundler in favor of BatchSpanProcessor + to comply w/ Specification

### LINKED ISSUE #100: Check if we need to make changes to support ForceFlush in batch processor
See recent change https://github.com/open-telemetry/opentelemetry-go/pull/1166 (issue: https://github.com/open-telemetry/opentelemetry-go/issues/1072)

### PR DESCRIPTION
**Pull request by jsuereth**
Fixes #100 

- Remove bundler and associated config/tests
- Remove metrics-options from trace-exporter config
- Update returned `flush` method from "pipeline" generation for `shutdown` method (only exposed sdk method in 1.0 spec)
- Add configuration for batch processor to Exporter-Pipeline helper method
- Update examples to denote `flush`=>`shutdown` semantic change

### CODE REVIEW

**Review by punya — APPROVED** [review]

File: exporter/trace/cloudtrace.go [hunk 1]
```diff
@@ -67,41 +67,14 @@ type options struct {
 	// Optional.
 	OnError func(err error)
 
-	// MonitoringClientOptions are additional options to be passed
-	// to the underlying Stackdriver Monitoring API client.
-	// Optional.
-	MonitoringClientOptions []option.ClientOption
```
Comment (dashpole) [inline exporter/trace/cloudtrace.go:73]: Should this just be renamed to TraceClientOptions instead of removed?  The actual option.ClientOption doesn't appear to be specific to GCM

File: exporter/trace/cloudtrace.go [hunk 2]
```diff
@@ -267,34 +171,47 @@ type Exporter struct {
...
 func NewExporter(opts ...Option) (*Exporter, error) {
 	o := options{Context: context.Background()}
 	for _, opt := range opts {
 		opt(&o)
 	}
+	return NewExporterWithOptions(&o)
+}
+
+// NewExporter creates a new Exporter thats implements trace.Exporter.
+// This version takes a single options struct.  This shouldn't be called
```
Comment (dashpole) [inline exporter/trace/cloudtrace.go:212]: If this shouldn't be called by user code, should this function be package-local (private)?

File: exporter/trace/trace.go [hunk 3]
```diff
@@ -62,72 +61,31 @@ func newTraceExporter(o *options) (*traceExporter, error) {
...
-	protoSpan := protoFromSpanSnapshot(sd, e.projectID, e.o.DisplayNameFormatter)
-	protoSize := proto.Size(protoSpan)
-	err := e.bundler.Add(protoSpan, protoSize)
-	e.checkBundlerError(err)
+func (e *traceExporter) ConvertSpan(_ context.Context, sd *export.SpanSnapshot) *tracepb.Span {
+	return protoFromSpanSnapshot(sd, e.projectID, e.o.DisplayNameFormatter)
+}
+
+func (e *traceExporter) Shutdown(ctx context.Context) error {
+	// return e.client.Close()
```
Comment (dashpole) [inline exporter/trace/trace.go:83]: nit: stray, commented out code.

File: exporter/trace/cloudtrace.go (hunk 1 above)
Comment (jsuereth) [reply exporter/trace/cloudtrace.go:73]: If you look right below this line, there already was a `TraceClientOptions` that had the exact same set of types.

File: exporter/trace/cloudtrace.go (hunk 2 above)
Comment (jsuereth) [reply exporter/trace/cloudtrace.go]: Oh my, just learned about capitals vs. not capitals.... will fix.

File: exporter/trace/trace.go (hunk 3 above)
Comment (jsuereth) [reply exporter/trace/trace.go]: Good catch.  I need to turn this one for reals, and the tests needed some tweaking (they double-close the connection)

File: exporter/trace/cloudtrace.go [hunk 4]
```diff
@@ -267,34 +171,44 @@ type Exporter struct {
...
-// for trace provider. Returns provider, flush function, and errors.
+// for trace provider. Returns provider, shutdown function, and errors.
 func NewExportPipeline(opts []Option, topts ...sdktrace.TracerProviderOption) (trace.TracerProvider, func(), error) {
-	exporter, err := NewExporter(opts...)
+	// TODO(suereth): Don't flesh options twice.
+	o := options{Context: context.Background()}
+	for _, opt := range opts {
+		opt(&o)
+	}
+	exporter, err := newExporterWithOptions(&o)
```
Comment (dashpole) [inline exporter/trace/cloudtrace.go:190]: `exporter, err := NewExporter(opts...)` seems simpler.  Why is this change needed?

File: exporter/trace/cloudtrace.go [hunk 5]
```diff
@@ -267,34 +171,44 @@ type Exporter struct {
...
 	if err != nil {
 		return nil, nil, err
 	}
-	tp := sdktrace.NewTracerProvider(append(topts, sdktrace.WithSyncer(exporter))...)
-	return tp, exporter.traceExporter.Flush, nil
+	tp := sdktrace.NewTracerProvider(
+		append(topts,
+			sdktrace.WithBatcher(exporter, o.BatchSpanProcessorOptions...))...)
+	return tp, func() {
```
Comment (dashpole) [inline exporter/trace/cloudtrace.go:197]: seems like `Shutdown` takes a context on purpose... Should we return a `func(context.Context)` instead of just a `func()`?

File: exporter/trace/cloudtrace.go (hunk 4 above)
Comment (jsuereth) [reply exporter/trace/cloudtrace.go:190]: I need access to the opts here, so I decided NOT to recreate them more than once in this flow.

File: exporter/trace/cloudtrace.go (hunk 5 above)
Comment (jsuereth) [reply exporter/trace/cloudtrace.go:197]: I think we evaluate that when we evaluate the convenience API.  I want to minimize churn in this change right now.  We have tickets to re-evaluate the convenience API.
